tests/foreman/api/test_templatesync.py

```diff
@@ -0,0 +1,1105 @@
+        self, module_org, create_import_export_local_dir
+    ):
+        """Assure templates with a given filter regex are not exported.
+
+        :id: 2f8ad8f3-f02b-4b2d-85af-423a228976f3
+
+        :Steps:
+            1. Using nailgun or direct API call
+               export templates matching that does not matches regex e.g: `robottelo`
+               using `negate` option.
+
+
+        :expectedresults:
+            1. Assert templates other than `robottelo` has been exported.
+
+        :CaseImportance: Medium
+        """
+        # Export some filtered templates to local dir
+        dir_name, dir_path = create_import_export_local_dir
+        entities.Template().exports(
+            data={
+                'repo': dir_path,
+                'organization_ids': [module_org.id],
+                'filter': 'ansible',
+                'negate': True,
+            }
+        )
+        assert ssh.command(f'find {dir_path} -type f -name *ansible* | wc -l').stdout[0] == '0'
+        assert ssh.command(f'find {dir_path} -type f | wc -l').stdout[0] != '0'
+
+    @tier2
+    def test_positive_export_and_import_with_metadata(
+        self, create_import_export_local_dir, module_org, module_location
+    ):
+        """Assure exported template contains metadata.
+
+        :id: ba8a34ce-c2c6-4889-8729-59714c0a4b19
+
+        :Steps:
+            1. Create a template in local directory and specify Org/Loc.
+            2. Use import to pull this specific template (using filter).
+            3. Using nailgun or direct API call
+               export the template containing metadata with Org/Loc specs
+               and specify the `metadata_export_mode` parameter
+               e.g: `-d {'metadata_export_mode': 'refresh'}` in POST body
+
+        :expectedresults:
+            1. Assert template is exported and new org/loc are present on
+               template metadata
+
+        :CaseImportance: Medium
+
+        :CaseAutomation: NotAutomated
+        """
+        ex_template = 'example_template.erb'
+        prefix = gen_string('alpha')
+        _, dir_path = create_import_export_local_dir
+        entities.Template().imports(
+            data={
+                'repo': dir_path,
+                'location_ids': [module_location.id],
+                'organization_ids': [module_org.id],
+                'prefix': prefix,
+            }
+        )
+        export_file = f'{prefix.lower()}{ex_template}'
+        # Export same template to local dir with refreshed metadata
+        entities.Template().exports(
+            data={
+                'metadata_export_mode': 'refresh',
+                'repo': dir_path,
+                'location_ids': [module_location.id],
+                'organization_ids': [module_org.id],
+                'filter': prefix,
+            }
+        )
+        result = ssh.command(
+            f'grep {module_org.name} ' f'{dir_path}/provisioning_templates/user_data/{export_file}'
```

Comment:
no need to separate the grep statement 
```suggestion
            f'grep {module_org.name}  {dir_path}/provisioning_templates/user_data/{export_file}'
```